src/routes/DigitalAssets/routes/Send/sagas/digitalAssetsSend.js

```diff
@@ -89,116 +88,209 @@ function* openView(action: ExtractReturn<typeof digitalAssetsSend.openView>): Sa
+      const gasPrice = getTransactionValue(userGasPrice, 18).toString()
+      yield put(digitalAssetsSend.setGasPriceValue(gasPrice))
+
+      const gasLimit = toBigNumber(userGasLimit).toString()
+      yield put(digitalAssetsSend.setGasLimitValue(gasLimit))
+    }
+  } else {
+    const txPriorityCoefficient: number = digitalAssetsSend.TXPRIORITY[priority]
+    if (!txPriorityCoefficient) {
+      yield put(digitalAssetsSend.setFormFieldError('gasPrice', 'Priority is not selected'))
+      return
+    }
+
+    const requestedGasPrice = yield* requestGasPrice()
+    if (!requestedGasPrice) {
+      yield put(digitalAssetsSend.setFormFieldError('gasPrice', 'Can\'t request gas price'))
+      return
+    }
+
+    const gasPrice = requestedGasPrice.times(txPriorityCoefficient).toString()
+    // save for the next step
+    yield put(digitalAssetsSend.setGasPriceValue(gasPrice))
+
+    if (!digitalAsset.isCustom &&
+      digitalAsset.blockchainParams &&
+      digitalAsset.blockchainParams.staticGasAmount) {
+      // save for the next step
+      const staticGasAmount = toBigNumber(digitalAsset.blockchainParams.staticGasAmount).toString()
+      yield put(digitalAssetsSend.setGasLimitValue(staticGasAmount))
+    } else {
+      // gas limit will be estimated at transaction send by jibrel-contracts-jsapi
+      yield put(digitalAssetsSend.setGasLimitValue(null))
+    }
+  }
+}
 
+function* checkAmount(digitalAsset: DigitalAsset): Saga<void> {
   const networkId: ExtractReturn<typeof selectCurrentNetworkId> =
     yield select(selectCurrentNetworkId)
 
-  const ownerAddress: ExtractReturn<typeof selectActiveWalletAddress> =
-    yield select(selectActiveWalletAddress)
-
   const currentBlock: ExtractReturn<typeof selectCurrentBlock> =
     yield select(selectCurrentBlock, networkId)
 
+  const ownerAddress: ExtractReturn<typeof selectActiveWalletAddress> =
+    yield select(selectActiveWalletAddress)
+
   const assetBalance: ExtractReturn<typeof selectBalanceByAssetAddress> = yield select(
     selectBalanceByAssetAddress,
     networkId,
     ownerAddress,
     currentBlock ? currentBlock.number.toString() : null,
-    address,
+    digitalAsset.blockchainParams.address
   )
 
-  if (!assetBalance || assetBalance.isError) {
-    return null
+  if (!assetBalance) {
+    return
   }
 
-  const balance: BigNumber = divDecimals(assetBalance.value, decimals)
+  const {
+    formFieldValues: {
+      amount,
+    },
+    gasSettings: {
+      gasPrice,
+      gasLimit,
+    },
+  }: ExtractReturn<typeof selectDigitalAssetsSend> = yield select(selectDigitalAssetsSend)
 
-  if (checkETH(address)) {
-    const fee: BigNumber = divDecimals(0, decimals)
-    const balanceWithFee: BigNumber = balance.plus(fee)
+  const amountToSend = getTransactionValue(amount, digitalAsset.blockchainParams.decimals)
+  const balance = toBigNumber(assetBalance.value)
+
+  const feeETH = (gasPrice && gasLimit)
+    ? toBigNumber(gasPrice).times(gasLimit)
+    : toBigNumber(0)
```

Comment:
we should never allow `feeETH = 0`
Because user will see that fee is zero, but when tx is mined, fee will be correct

Reply:
We need fetch the estimate gas for it, will be fixed in JWWEB-449, JWWEB-452